File: dart/8kyu/reversed_strings/bin/reversed_strings.dart
import "package:test/test.dart";

void main() {
  test('reversed world', () {
    expect(solution('world'), equals('dlrow'));
  });

  test('hello reversed', () {
    expect(solution('hello'), equals('olleh'));
  });

  test('reversed ', () {
    expect(solution(''), equals(''));
  });

  test('reversed h', () {
    expect(solution('h'), equals('h'));
  });
}

String solution(String str) {
  return String.fromCharCodes(str.runes.toList().reversed);
}

File: typescript/8kyu/reversed_strings/reversed_strings.ts
export function solution(str: string): string {
    return str.split('').reverse().join('');
}

// import the type of assertion library you wish to use (Chai recommended)
import { assert } from "chai";

describe("Sample Test Cases", function () {
    it("Should return a string", function () {
        assert.equal(solution('world'), 'dlrow');
        assert.equal(solution('hello'), 'olleh');
        assert.equal(solution(''), '');
        assert.equal(solution('h'), 'h');
    });
});